source/extensions/filters/common/fault/fault_config.h

```diff
@@ -70,31 +75,44 @@ class FaultAbortConfig {
 
     // Return the HTTP status code to use. Optionally passed HTTP headers that may contain the
     // HTTP status code depending on the provider implementation.
-    virtual absl::optional<Http::Code>
-    statusCode(const Http::RequestHeaderMap* request_headers) const PURE;
+    virtual absl::optional<Http::Code> httpStatusCode(const Http::HeaderEntry* header) const PURE;
```

Comment:
Why not continue to pass the full header map here and below? It seems like it would be more clear if the header was referenced closer to where it is used?